src/warpedview.jl

```diff
@@ -0,0 +1,59 @@
+immutable WarpedView{T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I} <: AbstractArray{T,N}
+    parent::A
+    transform::F1
+    transform_inv::F2
+    indices::I
+
+    function (::Type{WarpedView{T,N,A,F1,F2,I}}){T,N,A<:AbstractArray,F1<:Transformation,F2<:Transformation,I}(
+            parent::A, tform::F1, tinv::F2, indices::I)
+        @assert eltype(parent) == T
+        new{T,N,A,F1,F2,I}(parent, tform, tinv, indices)
+    end
+end
+
+function WarpedView{T,N,F<:Transformation}(inner::WarpedView{T,N}, tform_in::F)
+    tform = compose(tform_in, inner.transform)
+    tinv = inv(tform)
+    etp = inner.parent
+    inds = autorange(etp, tform)
+    WarpedView{T,N,typeof(etp),typeof(tform),typeof(tinv),typeof(inds)}(etp, tform, tinv, inds)
+end
+
+function WarpedView{T,N}(parent::AbstractArray{T,N}, args...)
+    itp = Interpolations.BSplineInterpolation{T,N,typeof(parent),Interpolations.BSpline{Interpolations.Linear},OnGrid,0}(parent)
```

Comment:
I create the interpolation this way, because of two orthogonal reasons. I'd be curious about your opinion on this.

1. `interpolate` calls `prefilter` (see: https://github.com/JuliaMath/Interpolations.jl/blob/master/src/b-splines/b-splines.jl#L63-L66) which causes the resulting `BSplineInterpolation` to be a decorator around a **copy** of `parent`. This goes a bit against the idea of a view. On the other hand I am guessing that there is a good reason to call `prefilter`.

2. A call to `interpolate` usually causes the resulting `BSplineInterpolation` to have a different element type. Given the meaning behind `warp` (that you were nice enough to explain to me here https://github.com/JuliaImages/ImageTransformations.jl/pull/13#issuecomment-292722621), I think we should be adamant about **only** performing a warp.